tests/test_marker.py

```diff
@@ -86,7 +82,9 @@ def test_dict_as_json(self):
         self.assertEqual(marker_dict, created_marker_dict)
 
     def test_many_as_ujson(self):
-        created_markers_dict = ujson.loads(ujson.dumps(self.markers))
+        created_markers_dict = ujson.loads(
+            ujson.dumps([m.as_dict() for m in self.markers])
```

Comment:
This change shouldn't be necessary?

Reply:
Please see the commit message for an explanation.

Reply:
One of the key reasons the method is `as_dict` is because of this auto discovery.

Interestingly I can reproduce this too.

Let's ship it like this and I can look into it properly during the marker rewrite